docs/ui/components/AskPageAI/AskPageAIChat.tsx

```diff
@@ -0,0 +1,440 @@
+import {
+  Children,
+  type CSSProperties,
+  type FormEvent,
+  type MouseEvent,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import ReactMarkdown, { type Components } from 'react-markdown';
+import remarkGfm from 'remark-gfm';
+
+import { FOOTNOTE } from '../Text';
+
+type AskPageAIChatProps = {
+  onClose: () => void;
+  onMinimize?: () => void;
+  pageTitle?: string;
+};
+
+type ConversationKey = string | null;
+
+export function AskPageAIChat({ onClose, onMinimize, pageTitle }: AskPageAIChatProps) {
+  const {
+    conversation,
+    submitQuery,
+    isGeneratingAnswer,
+    isPreparingAnswer,
+    error,
+    resetConversation,
+    stopGeneration,
+    addFeedback,
+  } = useChat();
+
+  const contextLabel = pageTitle?.trim() ? pageTitle : 'This page';
+  const [question, setQuestion] = useState('');
+  const [askedQuestions, setAskedQuestions] = useState<string[]>([]);
+  const previousTitleRef = useRef<string | undefined>(undefined);
+
+  useEffect(() => {
+    if (previousTitleRef.current !== pageTitle) {
+      previousTitleRef.current = pageTitle;
+      resetConversation();
+      setQuestion('');
+      setAskedQuestions([]);
+    }
+  }, [pageTitle, resetConversation]);
+
+  useEffect(() => {
+    if (conversation.length === 0 && askedQuestions.length > 0) {
+      setAskedQuestions([]);
+    }
+  }, [conversation.length, askedQuestions.length]);
+
+  const isBusy = isPreparingAnswer || isGeneratingAnswer;
+  const closeButtonThemeOverrides = useMemo(
+    () =>
+      ({
+        '--expo-theme-button-quaternary-hover': 'rgba(255,255,255,0.12)',
+        '--expo-theme-button-quaternary-text': '#ffffff',
+        '--expo-theme-button-quaternary-icon': '#ffffff',
+      }) as CSSProperties,
+    []
+  );
+  const headerAccentBackground = useMemo(() => ({ backgroundColor: 'rgba(255,255,255,0.1)' }), []);
+  const activeFeedbackBackground = useMemo(
+    () => ({ backgroundColor: 'rgba(255,255,255,0.12)' }),
+    []
+  );
+
+  const feedbackTarget = useMemo(() => {
+    for (let index = conversation.length - 1; index >= 0; index -= 1) {
+      const entry = conversation[index];
+      if (
+        entry &&
+        entry.id &&
+        'isFeedbackSubmissionEnabled' in entry &&
+        entry.isFeedbackSubmissionEnabled
+      ) {
+        return entry;
+      }
+    }
+    return null;
+  }, [conversation]);
+  const buildPrompt = useMemo(() => {
+    const origin = typeof window !== 'undefined' ? window.location.href : '';
+    return (text: string) =>
+      [
+        'You are ExpoDocsExpert, an assistant who answers questions strictly using the supplied Expo SDK documentation context.',
+        `The user is reading the Expo docs page titled "${contextLabel}" at ${origin || 'the latest Expo SDK docs'}.`,
+        'Only rely on the provided context. If the answer is missing, respond exactly with: "I couldn\'t find that on this page."',
```

Comment:
The "I don't know" prompt doesn't seem to work as expected. Asked about this in the BackgroundTask page.

<img width="452" height="485" alt="Image" src="https://github.com/user-attachments/assets/1c9d92cd-e4a4-427b-8eb5-201190405cef" />

Maybe we can ask it to reference the section it got the answer from (when possible), ideally so we could anchor link to it from the chat.